pkgs/tools/compression/zstd/default.nix

```diff
@@ -41,9 +37,10 @@ stdenv.mkDerivation rec {
 
   checkInputs = [ file ];
   doCheck = true;
-  preCheck = ''
+  checkPhase = ''
```

Comment:
When one defines `checkPhase` manually, shouldn't one include the running of the pre/post hooks so that other places' `override`s of e.g. `preCheck` still work?

Reply:
It seems that most overridden checkPhases do not run hooks, but some do. I have added runHooks.